pallets/marketplace/src/functions.rs

```diff
@@ -0,0 +1,69 @@
+use frame_support::{ensure, pallet_prelude::DispatchResult};
+use pallet_acurast_assets_manager::traits::AssetValidator;
+use reputation::BetaParameters;
+use sp_core::Get;
+
+use crate::{
+    AdvertisementFor, AdvertisementRestriction, Config, Error, Pallet, StoredAdvertisementPricing,
+    StoredAdvertisementRestriction, StoredReputation, StoredStorageCapacity,
+};
+
+impl<T: Config> Pallet<T> {
```

Comment:
Maybe move the other functions under `impl<T: Config> Pallet<T>` in `lib.rs` here too?